e2e/bitrise.yml

```diff
@@ -56,41 +61,104 @@ workflows:
     - TEST_APP_BRANCH: failing-ui-test
     - BITRISE_PROJECT_PATH: ios-simple-objc/ios-simple-objc.xcodeproj
     - BITRISE_SCHEME: ios-simple-objc
-    - XCODE_SIMULATOR_DEVICE: iPhone 8 Plus
+    - XCODE_SIMULATOR_DEVICE: iPhone 8
     - XCODE_SIMULATOR_OS_VERSION: "latest"
     - XCODE_SIMULATOR_PLATFORM: iOS Simulator
+    - SIGNLE_BUILD: "true"
+    - XCODE_OUTPUT_TOOL: xcodebuild
+    - RETRY_ON_FAILURE: "yes"
+    - EXPECT_TEST_FAILURE: "true"
+    - CACHE_LEVEL: "none"
     after_run:
     - _run
     - _check_outputs
     - _check_exported_artifacts
 
-  test_tvos:
+  test_parallel:
+    before_run:
+    - _expose_xcode_version
+    steps:
+    - script:
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -e
+
+            if [[ ${XCODE_MAJOR_VERSION} -lt 11 ]]; then
+              echo "This test case requires Xcode > 10, skipping..."
+              exit 0
+            fi
+
+            envman add --key XCODE_MAJOR_VERSION_GREATER_THAN_10 --value "true"
+
+            device="iPad Air (4th generation)"
+            if [[ ${XCODE_MAJOR_VERSION} -eq 10 ]]; then
+              device="iPad Air (3rd generation)"
+            elif [[ ${XCODE_MAJOR_VERSION} -eq 11 ]]; then
+              device="iPad Air (3rd generation)"
+            elif [[ ${XCODE_MAJOR_VERSION} -eq 12 ]]; then
+              device="iPad Air (4th generation)"
+            elif [[ ${XCODE_MAJOR_VERSION} -eq 13 ]]; then
+              device="iPad Air (4th generation)"
+            fi
+
+            echo "Xcode version based iPad device: $device"
+            envman add --key XCODE_SIMULATOR_DEVICE --value "$device"
+    - bitrise-run:
+        run_if: '{{enveq "XCODE_MAJOR_VERSION_GREATER_THAN_10" "true"}}'
+        inputs:
+        - workflow_id: utility_test_parallel
+        - bitrise_config_path: ./e2e/bitrise.yml
+
+  utility_test_parallel:
     envs:
-    - TEST_APP_URL: https://github.com/bitrise-io/sample-apps-tvos-swift.git
-    - TEST_APP_BRANCH: master
-    - BITRISE_PROJECT_PATH: NPO Live.xcworkspace
-    - BITRISE_SCHEME: NPO Live
-    - XCODE_SIMULATOR_DEVICE: Apple TV
+    - TEST_APP_URL: https://github.com/bitrise-io/sample-swift-project-with-parallel-ui-test.git
+    - TEST_APP_BRANCH: STEP-1054-test-plan
+    - BITRISE_PROJECT_PATH: BullsEye.xcworkspace
+    - BITRISE_SCHEME: BullsEye
     - XCODE_SIMULATOR_OS_VERSION: "latest"
-    - XCODE_SIMULATOR_PLATFORM: tvOS Simulator
+    - XCODE_SIMULATOR_PLATFORM: iOS Simulator
+    - SIGNLE_BUILD: "true"
+    - XCODE_OUTPUT_TOOL: xcodebuild
+    - RETRY_ON_FAILURE: "no"
+    - EXPECT_TEST_FAILURE: "false"
+    - CACHE_LEVEL: "swift_packages"
     after_run:
     - _run
     - _check_outputs
+    - _check_cache
 
-  test_parallel:
+  test_simulator_os_version:
+    before_run:
+    - _expose_xcode_version
+    steps:
+    - bitrise-run:
+        run_if: '{{enveq "XCODE_MAJOR_VERSION" "13"}}'
```

Comment:
It would make sense to check that Xcode version >= 13, rather then equal, to be compatible with future Xcode versions.

Reply:
Fixed in https://github.com/bitrise-steplib/steps-xcode-test/pull/168/commits/9ed0387c5cbe6f7936ade1b82dd7c3d4e2b561eb